File: pdui_flutter/lib/widgets/pdui_center.dart
import 'package:flutter/material.dart';

import '../core/pdui_expression_parser.dart';
import '../proto_out/pdui-proto-out.pb.dart';
import 'pdui_widget.dart';

class PduiCenter extends PduiWidget {
  PBExpression pbCenter;

  PduiCenter(this.pbCenter);

  @override
  Center resolve() {
    return Center(
      key: pbCenter.key != "" ? Key(pbCenter.key) : null,
      child: PduiExpressionResolver.parse(pbCenter.center.child),
    );
  }
}

File: pdui-express/lib/widgets/PDUICenter.ts
import { PDUIWidget } from "../models/PDUIWidget";
import type { PBCenter, PBExpression } from "../proto-out/pdui-proto-out.ts";

export class PDUICenter extends PDUIWidget {
    child: PDUIWidget;

    constructor({ key, child }: { key?: string; child: PDUIWidget }) {
        super({
            key,
            identity: "center",
            payload: { oneofKind: "center", center: child.toPB() },
        });

        this.child = child;
    }

    override toPB(): PBCenter {
        return {
            child: this.child.toPB(),
        };
    }

    override toExpression(): PBExpression {
        return {
            key: this.key,
            identity: this.identity,
            payload: {
                oneofKind: "center",
                center: { child: this.child.toExpression() },
            },
        };
    }
}
